### PR TITLE
Fix edge case assertion for FulfillAdvancedOrder fuzz tests

### PR DESCRIPTION
**Pull request by jameswenzel**
## Motivation
When `args.recipient` is `alice`, the `balanceOf` assertion will fail.

## Solution
Conditionally assert either numer or denom balance

### CODE REVIEW

File: test/foundry/FulfillAdvancedOrder.t.sol [hunk 1]
```diff
@@ -554,8 +549,11 @@ contract FulfillAdvancedOrder is BaseOrderTest {
             assertFalse(isCancelled);
             assertEq(totalFilled, context.args.numer);
             assertEq(totalSize, context.args.denom);
+            // if recipient is alice (offerer), final balance will be number minted, ie, denom
```
Comment (hrkrshnn) [inline test/foundry/FulfillAdvancedOrder.t.sol:552]: "number minted"?
